lnd.go

```diff
@@ -666,46 +666,64 @@ func bakeMacaroon(ctx context.Context, svc *macaroons.Service,
 	return mac.M().MarshalBinary()
 }
 
-// genMacaroons generates three macaroon files; one admin-level, one for
-// invoice access and one read-only. These can also be used to generate more
-// granular macaroons.
-func genMacaroons(ctx context.Context, svc *macaroons.Service,
-	admFile, roFile, invoiceFile string) error {
+// saveMacaroon bakes a macaroon with the specified macaroon permissions and
+// writes it to a file with the given filename and file permissions.
+func saveMacaroon(ctx context.Context, svc *macaroons.Service, filename string,
```

Comment:
micro nit: perhaps something like `bakeAndSaveMacaroon` or `genAndSaveMacaroon` would be more descriptive? 

Non-blocking though :)